core/services/relay/evm/parsed_types.go

```diff
@@ -0,0 +1,46 @@
+package evm
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/codec"
+	commontypes "github.com/smartcontractkit/chainlink-common/pkg/types"
+)
+
+type parsedTypes struct {
+	encoderDefs map[string]*CodecEntry
+	decoderDefs map[string]*CodecEntry
+}
+
+func (parsed *parsedTypes) toCodec() (commontypes.CodecTypeProvider, error) {
+	modByType := map[string]codec.Modifier{}
+	if err := addEntries(parsed.encoderDefs, modByType); err != nil {
+		return nil, err
+	}
+	if err := addEntries(parsed.decoderDefs, modByType); err != nil {
+		return nil, err
+	}
+
+	mod, err := codec.NewByItemTypeModifier(modByType)
+	if err != nil {
+		return nil, err
+	}
+	underlying := &evmCodec{
+		encoder:     &encoder{Definitions: parsed.encoderDefs},
+		decoder:     &decoder{Definitions: parsed.decoderDefs},
+		parsedTypes: parsed,
+	}
+	return codec.NewModifierCodec(underlying, mod, evmDecoderHooks...)
+}
+
+func addEntries(defs map[string]*CodecEntry, modByType map[string]codec.Modifier) error {
```

Comment:
just add a comment here that explains what happens here on a high level, this will help with understanding how chain reader works without diving deeper into codec code